packages/get-platform/src/getPlatform.ts

```diff
@@ -313,22 +348,22 @@ export async function getPlatform(): Promise<Platform> {
 
   // when the platform is linux
   if (platform === 'linux' && distro && libssl) {
-    return (distro + '-openssl-' + libssl) as Platform
+    return `${distro}-openssl-${libssl}` as Platform
   }
 
   // if just OpenSSL is known, fallback to debian with a specific libssl version
   if (libssl) {
-    return ('debian-openssl-' + libssl) as Platform
+    return `${defaultDistro}-openssl-${libssl}`
   }
 
   // if just the distro is known, fallback to latest OpenSSL 1.1
   if (distro) {
-    return (distro + '-openssl-1.1.x') as Platform
+    return `${distro}-openssl-${defaultLibssl}` as Platform
   }
 
   // use the debian build with OpenSSL 1.1 as a last resort
   // TODO: perhaps we should default to 'debian-openssl-3.0.x'
-  return 'debian-openssl-1.1.x'
+  return `${defaultDistro}-openssl-${defaultLibssl}`
```

Comment:
1. Unless I'm missing something, these four cases (lines 349-366) can this be simplified to

```ts
return `${distro ?? defaultDistory}-openssl-${libssl ?? defaultLibssl}` as Platform
```

2. Line 350 is the last one where we check that `platform === 'linux'`, but we still assume it afterwards. I feel like we are missing a warning similar to other ones added in this PR that we didn't recognise the operating system and assumed Linux as the default fallback.

Reply:
1. The whole cascade of `if`s in `getPlatform()` could be refactored / simplified, but I wouldn't risk doing that right now

2. Would you print a warning like `Prisma detected unsupported OS "${platform}, and may not work as expected. Defaulting to "linux".` after the latest `platform === 'linux'` check?

Reply:
2. Basically yes, I'd only slightly reword it: `Prisma detected unknown platform "${platform}", and may not work as expected. Defaulting to "linux".`. Known unsupported OSes should be covered by checks early in the function. I wonder if it should even be a hard error and not a warning.

Reply:
If we made this an error, and it worked with "linux" as a default, it'd be a breaking change in a non-major release

Reply:
Definitely not an error, warning would be fine.

I am ok with current state. Feel free to turn comments into comments or an issue to tackle later.